Astro CLI 1.7 Release Notes and New Docs (#1432)
astronomer/docs: astro/cli/release-notes.md

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.7.0

Release date: November 3, 2022 

## Deploy only DAGs to a Deployment with `astro deploy —dags`

User can now deploy just DAGs to their deployments with `astro deploy —dags`. Deploying DAGs is much faster than deploying an image and does not cause any down time for your Deployments. Read the documentation for this feature to learn how to enable it for your Deployments. 

> Additionally check out the new DAG Deploy CI/CD to enhance your CI/CD pipelines to take advantage of this new feature/

Reviewer: ```suggestion
To enhance your CI/CD pipelines, see <doc link>.
```

## Programmatically Query Deployment information with `astro deployment inspect`

> User can now return a JSON or YAML object of their Deployments that contain information about their Deployments current state and configuration. Check out the `astro deployment inspect` to see a full list of information fields that can be returned for your Deployments.

Reviewer: ```suggestion
User can now return a JSON or YAML object that contains information about the current state and configuration of a Deployment. See `astro deployment inspect` <doc link>.
```

> Users can also use the `—key` flag to return specific information about their deployment. For example `astro deployment inspect <deployment-id> -k information.status` will return `HEALTHY` for a healthy deployment.

Reviewer: ```suggestion
Users can use the `—key` flag to return specific information about a Deployment. For example, `astro deployment inspect <deployment-id> -k information.status` returns `HEALTHY` for a healthy deployment.
```

## Additional improvements

> - The commands `dev parse/pytest` have been refactored to have a clearer output. The docker container information no long shows and you will only see the output generate by pytest. These commands should also run faster now.

Reviewer: ```suggestion
- The commands `dev parse/pytest` have been refactored to run faster and provide clearer output. The Docker container information no longer appears and only the output generated by pytest is shown.
```

> - The `orginization switch` command now has a `—login-link` flag. This gives users a link to use to login if browser login does not work on their machine

Reviewer: ```suggestion
- The `orginization switch` command now has a `—login-link` flag to provide users with a login link when browser login doesn't work on their machine
```

> - The `orginization switch` command can now take either your organization name or ID as an argument

Reviewer: ```suggestion
- The `orginization switch` command now takes your Organization name or ID as an argument.
```

> - The `astro dev start` command will now timeout if the webserver does not become healthy. Users can use the `—wait` flag to give a custom time out amount.

Reviewer: ```suggestion
- The `astro dev start` command now times out if the webserver does not become healthy. Users can use the `—wait` flag to specify a custom time out interval.
```

## Bug fixes

- Fixed an issue where `astro deploy` with `colima` was failing due to a `registry login` issue

> - Fixed an issue were the `deployment list`  command was failing to display the Deployment’s Workspace ID

Reviewer: ```suggestion
- Fixed an issue were the `deployment list` command didn't display the Workspace ID for a Deployment
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.7.0

Release date: November 8, 2022 

## Deploy only DAGs with `astro deploy -—dags`

Use `astro deploy -—dags` with the Astro CLI to push only the `dags` directory of your Astro project to a Deployment on Astro. This is an alternative to rebuilding your Astro project's Docker image every time you change your code.

DAG-only deploys:

> - Are significantly faster than running `astro deploy`, as it avoids the process of building a Docker image and re-installing your dependencies.

Reviewer 2: ```suggestion
- Are significantly faster than running `astro deploy`.
```

Reviewer 2: This is already stated above

- Do not cause your workers to terminate and restart every time you need to make a change to a DAG, which leads to performance and cost benefits.

> - Enable teams to create separate CI/CD process for deploying DAGs and deploying other changes to your Astro project.

Reviewer 2: ```suggestion
- Enable teams to create separate CI/CD processes for deploying DAGs and deploying other changes to your Astro project.
```

> When you make changes to other files in your Astro project that aren't in the `dags` directory, the `astro deploy` command is still required. See [DAG-only deploys](deploy-code.md#dag-only-deploys) for more potential use cases and implementation details. 

Reviewer 2: ```suggestion
When you make changes to other files in your Astro project that aren't in the `dags` directory, the `astro deploy` command is still required. See [DAG-based deploys](deploy-code.md#dag-based-deploys) for more use cases and implementation details. 
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.7.0

Release date: November 8, 2022 

## Deploy only DAGs with `astro deploy -—dags`

> Use `astro deploy -—dags` with the Astro CLI to push only the `dags` directory of your Astro project to a Deployment on Astro. This is an alternative to the current image-only deployment method, which rebuilds all files in your Astro project into a Docker image every time you deploy your code to Astro.

Reviewer 3: ```suggestion
Use `astro deploy -—dags` with the Astro CLI to push only the `dags` directory of your Astro project to a Deployment on Astro. This is an alternative to the image-only deployment method, which rebuilds all files in your Astro project into a Docker image every time you deploy your code to Astro.
```

DAG-based deploys:

- Are significantly faster than running `astro deploy` when you only make changes to the `dags` directory.
- Do not cause your workers and schedulers to terminate and restart every time you make a change to a DAG, which can lead to both reliability and cost benefits.
- Enable your team to create separate CI/CD processes for deploying DAGs and deploying other changes to your Astro project.

When you make changes to other files in your Astro project that aren't in the `dags` directory, the `astro deploy` command is still required.

> To use this feature, you must enable it for each Deployment. See [DAG-based deploys](deploy-code.md#dag-based-deploys) for more information.

Reviewer 3: ```suggestion
To use this feature, you must enable it for each Deployment. See [DAG-based deploys](deploy-code.md#dag-based-deploys) for more information on use cases and implementation. For example CI/CD workflows with this feature enabled, see [CI/CD](ci-cd.md).
```

> To use this feature, you must enable it for each Deployment. See [DAG-based deploys](deploy-code.md#dag-based-deploys). For example CI/CD workflows with this feature enabled, see [CI/CD](ci-cd.md).

Reviewer 2: ```suggestion
To use this feature, you must enable it for each Deployment. See [DAG-only deploys](deploy-code.md#dag-only-deploys). For example CI/CD workflows with this feature enabled, see [CI/CD](ci-cd.md).
```

> To use this feature, you must enable it for each Deployment. See [DAG-based deploys](deploy-code.md#dag-based-deploys) for more information on use cases and implementation. For example CI/CD workflows with this feature enabled, see [CI/CD](ci-cd.md).

Reviewer 3: ```suggestion
To use this feature, you must enable it for each Deployment. See [DAG-based deploys](deploy-code.md#dag-based-deploys). For example CI/CD workflows with this feature enabled, see [CI/CD](ci-cd.md).
```

## New `astro deployment inspect` command

> You can now run `astro deployment inspect` to return a Deployment's current state and configuration as a JSON or YAML object. Use this command to quickly understand the state of your Deployment and as an alternative to viewing it in the Cloud UI.

Reviewer 3: ```suggestion
You can now run `astro deployment inspect` to return a Deployment's current state and configuration as a JSON or YAML object. This includes worker queue settings, Astro Runtime version, and more. Use this command to quickly understand the state of your Deployment as code and as an alternative to viewing it in the Cloud UI.
```

If you have any questions or a bug to report, contact [Astronomer support](https://cloud.astronomer.io/support).

## Astro CLI 1.7.0

> Release date: November 7, 2022 

Reviewer 3: ```suggestion
Release date: November 8, 2022 
```

## Deploy only DAGs with `astro deploy -—dags`

Use `astro deploy -—dags` to push only the `dags` directory of your Astro project to a Deployment on Astro. Deploying DAGs is significantly faster than deploying a Docker image and does not cause any down time for your Deployments. Additionally, the `--dags` flag can help you create separate CI/CD processes for DAG deployment and Astro project configuration. See [DAG-only deploys](deploy-code.md#dag-only-deploys) for more potential use cases and implementation details. 

> ## New command to programmatically query Deployment information

Reviewer 3: ```suggestion
## New `astro deployment inspect` command
```

> You can now run `astro deployment inspect` to return Deployment's current state and configuration as a JSON or YAML object.

Reviewer 3: ```suggestion
You can now run `astro deployment inspect` to return a JSON or YAML object that includes metadata of the Deployment, including its description and Astro Runtime version, as well as a running set of configurations. Use this command to quickly understand the state of your Deployment and as an alternative to viewing it in the Cloud UI.

The ability to use this YAML or JSON file with `astro deployment create` or `astro deployment update` to programmatically update Deployments defined as code is coming soon.
```

Reviewer 2: Cutting some of the details here and the "coming soon" statement, the reference guide covers all of this content and we shouldn't mention "Coming soon" unless there's something clearly amiss in the current behavior

For more information, see the [CLI command reference](cli/astro-deployment-inspect.md).

## Additional improvements

> - The outputs for `astro dev parse/pytest` no longer include Docker container logs for improved legibility.

Reviewer 3: ```suggestion
- The outputs for `astro dev parse` and `astro dev pytest` commands have improved legibility by no longer including Docker container logs.
```

> - The `astro organization switch` command now includes a `-—login-link` flag for logging in if browser login isn't available on your machine.

Reviewer 3: ```suggestion
- The `astro organization switch` command now includes a `-—login-link` flag that you can use to manually log in to a new or existing organization if you don't have a browser available on the same machine.
```

> - You can now provide either an Organization name or ID when running `astro organization switch` 

Reviewer 3: ```suggestion
- You can now provide either an Organization name or ID when running `astro organization switch` . Previously, you could only specify Organization ID.
```

Reviewer 2: Don't think previous behavior notes are relevant to include here, at least on this scope

> - `astro dev start` now times out if the webserver does not become healthy within a set period. Use the `-—wait` flag to specify a wait time.

Reviewer 3: ```suggestion
- `astro dev start` now times out after 1 minute if the Airflow webserver does not become healthy within a set period of time. For Apple M1 machines, it is 5 minutes. Use the `--wait` flag to specify a different wait time in seconds or minutes. This can help avoid your local Airflow environment hanging indefinitely.
```

Reviewer 2: Cut this down by a bit, assuming that if someone needs more info they will go to the reference guide or help text.

> - `astro dev start` now times out if the webserver does not become healthy in an expected amount of time. Use the `-—wait` flag to specify an amount of time to wait.

Reviewer: ```suggestion
- `astro dev start` now times out if the webserver does not become healthy within a set period. Use the `-—wait` flag to specify a wait time.
```

## Bug fixes

- Fixed an issue where `astro deploy` with `colima` was failing due to a `registry login` issue

> - Fixed an issue were the `deployment list` command didn't display the Workspace ID for a Deployment

Reviewer 3: ```suggestion
- Fixed an issue where the `astro deployment list` command didn't display the Workspace ID for a Deployment
```

## Astro CLI 1.6.1

> Release date: November 8, 2022 

Reviewer 3: Is this date wrong?

Reviewer 2: Must've gotten caught up in the commotion, fixed.
